components/TabbedView/TabbedView.jsx

```diff
@@ -159,86 +159,86 @@ const TabPanel = styled.div`
+function TabbedView({
+  children,
+  activeTab = undefined,
+  skin = 'neutral',
+  theme = {
+    components,
+    baseFontSize: defaultBaseFontSize,
+    breakpoints: defaultBreakpoints,
+    spacing: defaultSpacing,
+  },
+  fluid = false,
+  onTabClick = () => {},
+}) {
+  const [currentTab, setCurrentTab] = useState(activeTab);
+  const tabTitles = Children.toArray(children).map((tab) => tab.props.title);
+  function setDefaultTab() {
+    setCurrentTab(tabTitles[0]);
+  }
 
-    if (activeTab) {
-      this.state = { activeTab };
+  useEffect(() => {
+    if (tabTitles.includes(activeTab)) {
+      setCurrentTab(activeTab);
     } else {
-      const [
-        {
-          props: { title },
-        },
-      ] = Children.toArray(children);
-      this.state = { activeTab: title };
+      setDefaultTab();
     }
-  }
+  }, [activeTab]);
 
-  onTabClick = (tab, onTabClick) => {
-    this.setState({ activeTab: tab });
+  const handleTabClick = (tab) => {
+    setCurrentTab(tab);
     onTabClick(tab);
   };
 
-  sanitize = (str) =>
+  const sanitize = (str) =>
     str
       .normalize('NFD')
       .replace(/[\u0300-\u036f]/g, '')
       .replace(' ', '-')
       .toLowerCase();
 
-  render() {
-    const { children, skin, theme, fluid, onTabClick } = this.props;
-    const { activeTab } = this.state;
-
-    return (
-      <>
-        <Navbar theme={theme} skin={skin} fluid={fluid}>
-          {Children.map(children, ({ props: { title, badge, icon } }) => (
-            <NavItem
-              fluid={fluid}
-              quantityOfItems={Children.count(children)}
-              key={title}
-              onClick={() => this.onTabClick(title, onTabClick)}
-              skin={skin}
-              theme={theme}
-              id={`${this.sanitize(title)}-tab`}
-              aria-controls={`${this.sanitize(title)}-panel`}
-              aria-selected={title === activeTab}
-            >
-              {icon && <IconContainer>{icon}</IconContainer>}
-              {title && <TitleContainer>{title}</TitleContainer>}
-              {badge && <BadgeContainer>{badge}</BadgeContainer>}
-            </NavItem>
-          ))}
-        </Navbar>
-
-        {Children.map(
-          children,
-          ({ props: { title, children: tabContent } }) => (
-            <RenderIf conditional={title === activeTab}>
-              <TabPanel
-                role="tabpanel"
-                key={`${this.sanitize(title)}-panel-key`}
-                id={`${this.sanitize(title)}-panel`}
-                aria-labelledby={`${this.sanitize(title)}-tab`}
-              >
-                {tabContent}
-              </TabPanel>
-            </RenderIf>
-          ),
-        )}
-      </>
-    );
-  }
+  return (
+    <>
+      <Navbar theme={theme} skin={skin} fluid={fluid}>
+        {Children.map(children, ({ props: { title, badge, icon } }) => (
+          <NavItem
+            fluid={fluid}
+            quantityOfItems={Children.count(children)}
```

Comment:
Taking this opportunity, could you add this method at the top of the component (after the tabTitles declaration)? I believe that the component will have more performance because this count only needs to be done once, and not on each item.